modules/programs/vscode/haskell.nix

```diff
@@ -1,62 +1,33 @@
 { pkgs, config, lib, ... }:
 
-with lib;
+let cfg = config.programs.vscode.haskell;
 
-let
-
-  cfg = config.programs.vscode.haskell;
-
-  defaultHieNixExe = hie-nix.hies + "/bin/hie-wrapper";
-  defaultHieNixExeText =
-    literalExpression ''"''${pkgs.hie-nix.hies}/bin/hie-wrapper"'';
-
-  hie-nix = pkgs.hie-nix or (abort ''
-    vscode.haskell: pkgs.hie-nix missing. Please add an overlay such as:
-    ${exampleOverlay}
-  '');
-
-  exampleOverlay = ''
-    nixpkgs.overlays = [
-      (self: super: { hie-nix = import ~/src/hie-nix {}; })
-    ]
-  '';
-
-in {
+in with lib; {
   options.programs.vscode.haskell = {
     enable = mkEnableOption "Haskell integration for Visual Studio Code";
 
-    hie.enable = mkOption {
```

Comment:
would be nice to add some warning instead so that it doesnt straightaway breaks configs

Reply:
Good point, will do

Reply:
Done